### ISSUE TITLE
bug(py): Fix nnet analogous numerical precision — orbital produces constant output for nnet binary/multiclass classification

### ISSUE THREAD

**Issue by davidrsch** [davidrsch/orbital-py-fork#21]
## Summary\n\nThe R `nnet` binary and multiclass classification probability tests are failing with a consistent symptom: `predict(orbital_obj, data)` returns **constant probability values** regardless of the input features. For example, `.pred_0 = 0.6` for every row in the binary case, and `.pred_setosa ≈ 1.0` for every row in the multiclass case.\n\nThis is a bug in `R/model-nnet.R`'s weight extraction — the hidden and/or output layer weight matrices are likely being extracted incorrectly from `x$wts`, causing the SQL expression to not depend on the input columns.\n\n## Symptoms\n\n```\nFailure ('test-model-nnet.R:66:5'): mlp() nnet works with binary prob\n  actual$.pred_0  | expected$.pred_0\n  0.600           | 0.550   # expected varies per row, actual is constant\n\nFailure ('test-model-nnet.R:96:5'): mlp() nnet works with multiclass prob\n  actual$.pred_setosa | expected$.pred_setosa\n  1.00                | 0.58   # nearly all probability mass on one class\n```\n\n## Root Cause Hypothesis\n\nIn `R/model-nnet.R`, the weight extraction loop for hidden/output layers uses:\n```r\nfor (i in seq_len(n_h)) {\n    base <- (i - 1L) * (n_in + 1L)\n    hidden_b[i] <- x$wts[base + 1L]\n    for (j in seq_len(n_in)) {\n        hidden_w[i, j] <- x$wts[base + j + 1L]\n    }\n}\n```\nThis indexing matches the nnet packing convention (`bias, w1, w2, ...` per hidden unit). If the constant-output bug is that all input weights are zero, either:\n1. The indexing is off-by-one in a way that reads bias coefficients into weight slots\n2. The `x$wts` vector is ordered differently than expected for parsnip-wrapped `nnet` vs. bare `nnet`\n3. The `n_in` doesn't account for a bias column that nnet adds to the input\n\nVerify against `nnet::nnet`'s documented weight layout.\n\n## Acceptance Criteria\n\n- `test-model-nnet.R:66` (binary prob) and `test-model-nnet.R:96` (multiclass prob) both pass\n- `predict(orbital(fit, type=\"prob\"), data)` matches `predict(fit, data, type=\"raw\")` within tolerance 1e-5\n- The weight extraction logic is documented with a comment explaining the `nnet` packing format\n\n## Notes\n\n- `nnet` uses a single `$wts` vector; layout: `[bias_h1, w_h1_1..w_h1_n, bias_h2, ..., bias_o1, w_o1_1..w_o1_h]`\n- `x$n` gives `[n_input, n_hidden, n_output]`\n- `x$coefnames` gives the input column names (without intercept)\n- The parsnip `fit` wrapper stores the nnet object in `fit$fit$fit` — verify the correct extraction path when accessed through parsnip",

**Comment by davidrsch** [comment]
✅ Investigated in [`feat/ann-mlp-support`](https://github.com/davidrsch/orbital-1/tree/feat/ann-mlp-support) (commit [500e023](https://github.com/davidrsch/orbital-1/commit/500e023)).

This issue was the Python-side analog of R bug davidrsch/orbital#22. The root cause in R was confirmed: `model-nnet.R` was reading bias slots into weight positions. The Python orbital package does not have a nnet-specific translator (sklearn `MLPClassifier` uses the standard Gemm/Relu/Softmax ONNX path and is correct). No Python code changes were needed.

A test was added to `tests/test_pytorch_mlp.py` confirming sklearn MLP predictions match orbital within tolerance 1e-5.

For the fixed R implementation: davidrsch/orbital#22.
